nvc/listFiles.go

```diff
@@ -0,0 +1,28 @@
+package nvc
+
+import (
+	"io"
+)
+
+func ListEntries(r io.Reader) ([]nvcEntry, error) {
+	if ok, magicErr := readNVCMagic(r); !ok {
+		return nil, magicErr
+	}
+
+	count, countErr := readCount(r)
+	if countErr != nil {
+		return nil, countErr
+	}
+
+	entries := make([]nvcEntry, count)
+
+	for i := range(entries) {
+		entry, eErr := readNvcEntry(r)
+		if eErr != nil {
+			return nil, eErr
+		}
+		entries[i] = entry
+	}
```

Comment:
From a testing perspective, would be easier to test interfaces that operate on `io.Reader`?  This could also allow for composing functions as an easy way to handle edge cases of the format, such as format changes etc.

Reply:
Maybe keep the existing separate functions but add an additional helper function that just calls all of them in the correct order?

Reply:
good idea, we should work towards that